### PR TITLE
Parameterize quill and idl2json

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
We would like to use specific versions of quill and idl2json, but to update these regularly.  To that end we would like to keep versions in a single location.

# Changes
- Specify the `idl2json` and `quill` versions in `versions.bash` along with other tools.
- Make the idl2json and quill installers accept `--version pinned`, like other `snsdemo` installers.
- Use the pinned version.

# Tests
- The `--version` pinned flag is exercised in `CI`.
- I have also used the `pinned` and `latest` versions locally.  I note that the latest `quill` does not run on the IC's production OS (Ubuntu 20.04).

### CODE REVIEW

File: bin/dfx-software-idl2json-install [hunk 1]
```diff
@@ -2,17 +2,20 @@
...
 PATH="$SOURCE_DIR:$PATH"
+
 . "$SOURCE_DIR/versions.bash"
+PINNED_VERSION="$IDL2JSON_VERSION"
 
 # Source the clap.bash file ---------------------------------------------------
 source "$SOURCE_DIR/clap.bash"
 # Define options
-clap.define short=v long=version desc="The release to install" variable=IDL2JSON_VERSION default="latest"
+clap.define short=v long=version desc="The release to install.  Options: Semantic version, latest, pinned." variable=IDL2JSON_VERSION default="latest"
```
Comment (dskloetd) [inline bin/dfx-software-idl2json-install:12]: It seems like making `"pinned"` the default would result in fewer surprises. Shall we make "pinned" the default?
The following all have `default="pinned"`:
```
bin/dfx-software-didc-install
bin/dfx-software-didc-url
bin/dfx-software-internet-identity-ci-wasm-url
bin/dfx-software-internet-identity-install
bin/dfx-software-nns-dapp-ci-wasm-url
bin/dfx-software-sns-aggregator-ci-wasm-url
bin/dfx-software-sns-aggregator-install
```

File: bin/dfx-software-quill-install [hunk 2]
```diff
@@ -2,17 +2,20 @@
...
 PATH="$SOURCE_DIR:$PATH"
+
 . "$SOURCE_DIR/versions.bash"
+PINNED_VERSION="${QUILL_VERSION}"
 
 # Source the clap.bash file ---------------------------------------------------
 source "$SOURCE_DIR/clap.bash"
 # Define options
-clap.define short=v long=version desc="The release to install" variable=QUILL_VERSION default="latest"
+clap.define short=v long=version desc="The release to install.  Options: Semantic version, latest, pinned." variable=QUILL_VERSION default="latest"
```
Comment (dskloetd) [inline bin/dfx-software-quill-install:12]: Shall we make "pinned" the default?

**Review by dskloetd — APPROVED** [review]

File: bin/dfx-software-idl2json-install (hunk 1 above)
Comment (bitdivine) [reply bin/dfx-software-idl2json-install:12]: Yes, we can do.  Changing the default may break some code but we know most of the users!

File: bin/dfx-software-quill-install (hunk 2 above)
Comment (bitdivine) [reply bin/dfx-software-quill-install]: Done

**Pull request merged**